src/main/java/org/openlmis/notification/web/usercontactdetails/UserContactDetailsController.java

```diff
@@ -169,6 +176,53 @@ public UserContactDetailsDto saveUpdateUserContactDetails(
     return toDto(contactDetails);
   }
 
+  /**
+   * Creates or updates contact details for multiple users.
+   *
+   * @param userContactDetailsDtoList  list of user contact details
+   * @return  the new or updated contact details for list of users
+   */
+  @PutMapping("/userContactDetails/batch")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public UserContactDetailsResponseDto saveUsersContactDetails(
+      @RequestBody List<UserContactDetailsDto> userContactDetailsDtoList) {
+    List<UserContactDetailsResponseDto.UserDetailsResponse> successfulResults = new ArrayList<>();
+    List<UserContactDetailsResponseDto.FailedUserDetailsResponse> failedResults = new ArrayList<>();
+
+    for (UserContactDetailsDto dto : userContactDetailsDtoList) {
+      try {
+        BindingResult bindingResult = new BeanPropertyBindingResult(dto, "userContactDetailsDto");
+        permissionService.canManageUserContactDetails(dto.getReferenceDataUserId());
```

Comment:
This might be slow, can you look into this? Can we run a single check or permission to Manage users instead checking it for each individual user? 
I've had a quick look at the `canManageUserContactDetails` method and it seems it does **TWO** requests to reference-data service to check the single right.

Reply:
This is actually not a **TWO** requests and it is not slow. 
I think the call is made without authentication during import, it's that `authentication.isClientOnly()` branch of the `canManageUserContactDetails` method.